### DISCUSSION TITLE
Wiring up a <select> component with multiple values

### DISCUSSION THREAD

**Discussion by hornta** [react-hook-form/react-hook-form#8018] [Q&A] [answered]
Hi!

This question is more of a "am I doing this right or are there a better way" than anything else :-)

I had to implement this "multiselect" type of component where I display the selected options outside the form element. I can't help but feel like the amount of code to get this to work correctly are a bit too much. Are there a better way to implement it like I did? 

https://codesandbox.io/s/modest-sid-fe3ei6?file=/src/App.tsx

I appreciate suggestions and improvements

**Comment by Moshyfawn** [comment, marked as answer]
Hey, @hornta!

I'd personally suggest to make the input and the list of selected values into one component to isolated the re-renders and conveniently provide the field handlers and it's value to both the select and the list.

To avoid manually triggering the validation, you can set the validation `mode: "onChange"`

See [this rough CSB](https://codesandbox.io/s/festive-cannon-dw0unh?file=/src/App.tsx) I hacked together for example.
